src/java.desktop/share/classes/javax/swing/text/html/CSS.java

```diff
@@ -1405,307 +1406,164 @@ else if (str.startsWith("rgba("))
-        else if (str.equalsIgnoreCase("Peru"))
-            color = hexToColor("#cd853f");
-        else if (str.equalsIgnoreCase("Pink"))
-            color = hexToColor("#ffc0cb");
-        else if (str.equalsIgnoreCase("Plum"))
-            color = hexToColor("#dda0dd");
-        else if (str.equalsIgnoreCase("Powderblue"))
-            color = hexToColor("#b0e0e6");
-        else if (str.equalsIgnoreCase("Rebeccapurple"))
-            color = hexToColor("#663399");
-        else if (str.equalsIgnoreCase("Rosybrown"))
-            color = hexToColor("#bc8f8f");
-        else if (str.equalsIgnoreCase("Royalblue"))
-            color = hexToColor("#4169e1");
-        else if (str.equalsIgnoreCase("Saddlebrown"))
-            color = hexToColor("#8b4513");
-        else if (str.equalsIgnoreCase("Salmon"))
-            color = hexToColor("#fa8072");
-        else if (str.equalsIgnoreCase("Sandybrown"))
-            color = hexToColor("#f4a460");
-        else if (str.equalsIgnoreCase("Seagreen"))
-            color = hexToColor("#2e8b57");
-        else if (str.equalsIgnoreCase("Seashell"))
-            color = hexToColor("#fff5ee");
-        else if (str.equalsIgnoreCase("Sienna"))
-            color = hexToColor("#a0522d");
-        else if (str.equalsIgnoreCase("Skyblue"))
-            color = hexToColor("#87ceeb");
-        else if (str.equalsIgnoreCase("Slateblue"))
-            color = hexToColor("#6a5acd");
-        else if (str.equalsIgnoreCase("Slategray"))
-            color = hexToColor("#708090");
-        else if (str.equalsIgnoreCase("Slategrey"))
-            color = hexToColor("#708090");
-        else if (str.equalsIgnoreCase("Snow"))
-            color = hexToColor("#fffafa");
-        else if (str.equalsIgnoreCase("Springgreen"))
-            color = hexToColor("#00ff7f");
-        else if (str.equalsIgnoreCase("Steelblue"))
-            color = hexToColor("#4682b4");
-        else if (str.equalsIgnoreCase("Tan"))
-            color = hexToColor("#d2b48c");
-        else if (str.equalsIgnoreCase("Thistle"))
-            color = hexToColor("#d8bfd8");
-        else if (str.equalsIgnoreCase("Tomato"))
-            color = hexToColor("#ff6347");
-        else if (str.equalsIgnoreCase("Turquoise"))
-            color = hexToColor("#40e0d0");
-        else if (str.equalsIgnoreCase("Violet"))
-            color = hexToColor("#ee82ee");
-        else if (str.equalsIgnoreCase("Wheat"))
-            color = hexToColor("#f5deb3");
-        else if (str.equalsIgnoreCase("Whitesmoke"))
-            color = hexToColor("#f5f5f5");
-        else if (str.equalsIgnoreCase("Yellowgreen"))
-            color = hexToColor("#9acd32");
-        else
+        else if ((color = colorNamed.get(str.toLowerCase(Locale.ROOT))) == null)
             color = hexToColor(str); // sometimes get specified without leading #
         return color;
     }
 
+    private static TreeMap<String, Color> colorNamed = new TreeMap<String, Color>(
```

Comment:
Why a `TreeMap` instead of just using `Map.ofEntries(…)`?

--------------------------------------------------------------------------------

The latter is unmodifiable and its `table` is marked as `@Stable`[^1], which means that the JIT is allowed to treat it as fully constant[^2].

[^1]: https://github.com/openjdk/jdk/blob/ea2c82e74f5580f396920f9e561cbec80c03f373/src/java.base/share/classes/java/util/ImmutableCollections.java#L1164-L1170
[^2]: https://github.com/openjdk/jdk/blob/ea2c82e74f5580f396920f9e561cbec80c03f373/src/java.base/share/classes/jdk/internal/vm/annotation/Stable.java#L64-L72

Reply:
@ExE-Boss I choosed a TreeMap because it seems it implements the best algorithm to acces algorithm to the values log(n) time.
By maintaining the order of its elements, it's can use compare to accelerate time access.

Reply:
@ExE-Boss Map 35% faster than TreeMap.